src/components/cookie-consent-banner.tsx

```diff
@@ -14,11 +15,55 @@ export const CookieConsentBanner = () => {
       return;
     }
 
+    const GEO_CACHE_KEY = "geo:is_eu";
+    const GEO_CACHE_TTL_MS = 24 * 60 * 60 * 1000; // 24 hours
+
+    const cacheSchema = z.object({
+      isEuropean: z.boolean(),
+      expiresAt: z.number().int().nonnegative(),
+    });
+
+    function getCachedIsEuropean(): boolean | null {
+      try {
+        const raw = localStorage.getItem(GEO_CACHE_KEY);
+        if (!raw) return null;
+        const parsed = cacheSchema.safeParse(JSON.parse(raw));
+        if (!parsed.success) {
+          localStorage.removeItem(GEO_CACHE_KEY);
+          return null;
```

Comment:
I feel like it's better to add a new field to the useGlobalStore instead of defining a new entry in local storage